core/parameters/src/config.rs

```diff
@@ -94,3 +98,111 @@ impl Default for AccountCreationConfig {
+    pub max_outgoing_gas: Gas,
+
+    /// The minimum gas each shard can send to a shard that is not fully congested.
+    ///
+    /// The actual gas forwarding allowance is a linear interpolation between
+    /// [`MIN_OUTGOING_GAS`] and [`MAX_OUTGOING_GAS`], or 0 if the receiver is
+    /// fully congested.
+    pub min_outgoing_gas: Gas,
+
+    /// How much gas the chosen allowed shard can send to a 100% congested shard.
+    ///
+    /// This amount is the absolute minimum of new workload a congested shard has to
+    /// accept every round. It ensures deadlocks are provably impossible. But in
+    /// ideal conditions, the gradual reduction of new workload entering the system
+    /// combined with gradually limited forwarding to congested shards should
+    /// prevent shards from becoming 100% congested in the first place.
+    pub red_gas: Gas,
+
+    /// The maximum amount of gas in a chunk spent on converting new transactions to
+    /// receipts.
+    ///
+    /// The actual gas forwarding allowance is a linear interpolation between
+    /// [`MIN_TX_GAS`] and [`MAX_TX_GAS`], based on the incoming congestion of the
+    /// local shard. Additionally, transactions can be rejected if the receiving
+    /// remote shard is congested more than [`REJECT_TX_CONGESTION_THRESHOLD`] based
+    /// on their general congestion level.
+    pub max_tx_gas: Gas,
+
+    /// The minimum amount of gas in a chunk spent on converting new transactions
+    /// to receipts, as long as the receiving shard is not congested.
+    ///
+    /// The actual gas forwarding allowance is a linear interpolation between
+    /// [`MIN_TX_GAS`] and [`MAX_TX_GAS`], based on the incoming congestion of the
+    /// local shard. Additionally, transactions can be rejected if the receiving
+    /// remote shard is congested more than [`REJECT_TX_CONGESTION_THRESHOLD`] based
+    /// on their general congestion level.
+    pub min_tx_gas: Gas,
+
+    /// How much congestion a shard can tolerate before it stops all shards from
+    /// accepting new transactions with the receiver set to the congested shard.
+    pub reject_tx_congestion_threshold: f64,
+}
+
+// The following default constants have been defined in
+// [NEP-539](https://github.com/near/NEPs/pull/539) after extensive fine-tuning
+// and discussions.
+
+impl Default for CongestionControlConfig {
+    fn default() -> Self {
+        const PGAS: Gas = 10u64.pow(15);
+        const TGAS: Gas = 10u64.pow(12);
+
+        Self {
+            max_congestion_incoming_gas: 20 * PGAS,
+            max_congestion_outgoing_gas: 2 * PGAS,
+            max_congestion_memory_consumption: bytesize::ByteSize::mb(1000u64).0,
+            max_congestion_missed_chunks: 10,
+            max_outgoing_gas: 300 * PGAS,
+            min_outgoing_gas: 1 * PGAS,
+            red_gas: 1 * PGAS,
+            max_tx_gas: 500 * TGAS,
+            min_tx_gas: 20 * TGAS,
+            reject_tx_congestion_threshold: 0.25,
+        }
+    }
+}
```

Comment:
This might make sense to have for certain tests. But I recommend to avoid using it in production. It can be very confusing to have a dozen different ways of setting a default value.

Instead, the parameters should be added to `core/parameters/res/runtime_configs/parameters.yaml` (+ `parameters_testnet.yaml`). The values don't matter much, as they are unused. I would probably set them to 0 or whatever the equivalent of "no congestion control" would be.

And then in `core/parameters/res/runtime_configs/142.yaml` you should define the real values and out the comment about how we derived the values there.

Reply:
Good point, removed. In some places I still rely on the defaults for CongestionInfo and ExtendedCongestionInfo - let me know if you think we should also get rid of those.

Reply:
Looks good to me now, no problem IMO to use defaults for CongestionInfo and ExtendedCongestionInfo in the remaining places.